### PR TITLE
Release idle connections fixes #33

### LINKED ISSUE #33: Connection Pool is not releasing idle connections
It seems like the driver opens connections until `maxPoolSize` is reached but never close a connection. So when the `maxPoolSize` is 10 I get 10 open connections which are all idle when nothing is using the application. I think the driver/module should release unused connection, so the Database-Server doesn't have tons of open idle connections.

 I'm using PostgreSQL btw.

### PR DESCRIPTION
**Pull request by adamdickmeiss**
New configuration connectionReleaseDelay which has a default of 1 minute.

### CODE REVIEW

File: src/main/java/io/vertx/ext/asyncsql/impl/pool/AsyncConnectionPool.java [hunk 1]
```diff
@@ -41,25 +43,29 @@
...
   private final Deque<Connection> availableConnections = new ArrayDeque<>();
   private final Deque<Handler<AsyncResult<Connection>>> waiters = new ArrayDeque<>();
+  private final Map<Connection,Long> timers = new HashMap<>();
 
   public AsyncConnectionPool(Vertx vertx, JsonObject globalConfig, Configuration connectionConfig) {
     this.vertx = vertx;
     this.maxPoolSize = globalConfig.getInteger("maxPoolSize", DEFAULT_MAX_POOL_SIZE);
     this.maxConnectionRetries = globalConfig.getInteger("maxConnectionRetries", DEFAULT_MAX_CONNECTION_RETRIES);
     this.connectionRetryDelay = globalConfig.getInteger("connectionRetryDelay", DEFAULT_CONNECTION_RETRY_DELAY);
+    this.connectionReleaseDelay = globalConfig.getInteger("connectionReleaseDelay", DEFAULT_CONNECTION_RELEASE_DELAY);
```
Comment (Narigo) [inline src/main/java/io/vertx/ext/asyncsql/impl/pool/AsyncConnectionPool.java]: As you have `0` as another special value - why not use that in order to not break code relying on over 1 minute?

**Comment by adamdickmeiss** [comment]
On Tue, Jun 19, 2018 at 10:54 AM Joern Bernhardt <notifications@github.com>
wrote:

> *@Narigo* requested changes on this pull request.
>
> Thanks for helping out @adamdickmeiss <https://github.com/adamdickmeiss> !
>
>    1. Can you please add tests to see if the connections really get
>    closed after the set amount of time?
>    2. Can you please document the new option and especially the special "0
>    or less" meaning "no idle timeout"
>
> OK.. Tried to add a test and some documentation.


https://github.com/folio-org/vertx-mysql-postgresql-client/commits/release-connection-pool-3-6-0

I also changed default behavior to 0 = no releasing.

But I want to stress that applications are probably faulty if they rely on
"released" connections being kept alive.

The pool (reuse) is pure optimization , unless I'm seriously mistaking
things :-)


>
>    1. Did you sign the Eclipse contribution agreement? See the Vert.x
>    wiki about contributions
>    <https://github.com/eclipse/vert.x/blob/master/CONTRIBUTING.md#contributions>
>
> Signed.

Let me know if I should squash the changes into a new commit or something..

/ Adam

**Comment by adamdickmeiss** [comment]
Any news on this? Should I make a new pull request with one big commit?

**Comment by julianladisch** [comment]
@Narigo Please review adamdickmeiss' latest changed. What else should we improve?

**Pull request closed without merging**